{"cells":[{"cell_type":"markdown","metadata":{},"source":["# Essais de requêtes"]},{"cell_type":"markdown","metadata":{},"source":["## Divisions et URLs"]},{"cell_type":"code","execution_count":10,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["['EnsemblMetazoa', 'EnsemblFungi', 'EnsemblPlants', 'EnsemblProtists', 'EnsemblVertebrates', 'EnsemblBacteria']\n"]}],"source":["# !/usr/bin/ python3\n","#-*- coding : utf-8 -*- \n","import requests, sys, json\n"," \n","server = \"https://rest.ensembl.org\"\n","ext_div = \"/info/divisions?\"\n"," \n","d = requests.get(server+ext_div, headers={ \"Content-Type\" : \"application/json\"})\n"," \n","if not d.ok:\n","  d.raise_for_status()\n","  sys.exit()\n"," \n","divs = d.json() #liste avec les divisions\n","print(divs)\n","\n"]},{"cell_type":"markdown","metadata":{},"source":["## Accession\n","Aller chercher un numéro d'accession pour l'espèce qui servira d'entrée pour trouver la BdD ensembl appropriée"]},{"cell_type":"code","execution_count":17,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["vertebrates\n"]}],"source":["# !/usr/bin/ python3\n","#-*- coding : utf-8 -*- \n","import requests, sys\n","\n","def RetrieveDiv (species) :\n","    # Requête des infos concernant l'espèce\n","    server = \"https://rest.ensembl.org\" # Serveur Ensembl pour requête API\n","\n","    ext = f\"/info/assembly/{species}?\" # Requête API\n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"}) \n","\n","    # Retour en cas d'erreur\n","    if not r.ok:\n","        r.raise_for_status()\n","        sys.exit() \n","        \n","    decoded = r.json() # Résultat API\n","    accession = decoded['assembly_accession'] # Récupération d'une accession d'un génome    \n","\n","    # Requête des infos concernant l'accession\n","    ext = f\"/info/genomes/assembly/{accession}?\"\n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"}) \n","\n","    # Retour en cas d'erreur\n","    if not r.ok:\n","        r.raise_for_status()\n","        sys.exit()\n","    decoded = r.json() # Résultat API\n","\n","    if decoded[\"division\"][7:].lower() == \"vertebrates\" :\n","        div = \"www.\"\n","    else : \n","        div = decoded[\"division\"][7:].lower()\n","    \n","    return(div)\n","\n","def InfoGene (species, symbol) :\n","    # Serveur Ensembl pour requête API\n","    server = \"https://rest.ensembl.org\"\n","\n","    # Lien vers l'espèce et symbole de gène courants\n","    ext = f\"/lookup/symbol/{species}/{symbol}?expand=1\" \n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n","    decoded = r.json()\n","\n","    # Retour en cas d'erreur d'info de l'API\n","    if not r.ok:\n","        r.raise_for_status()\n","        sys.exit()\n","\n","    # Construction du dictionnaire contenant les infos fomatées pour le html\n","    species_info = {}\n","    species_info[\"gene_symbol\"] = symbol\n","    species_info[\"gene_id\"] = decoded[\"id\"]\n","    species_info[\"division\"] = RetrieveDiv(species)\n","\n","    for i in decoded[\"Transcript\"] :\n","        if \"transcript_id\" not in species_info :\n","            species_info[\"transcript_id\"] = i[\"id\"]\n","            species_info[\"prot_id\"] = i[\"Translation\"][\"id\"]\n","        else : \n","            species_info[\"transcript_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"id\"]\n","            species_info[\"prot_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"Translation\"][\"id\"]\n","\n","    species_info[\"gene_browser\"] = f\"http://{species_info[\"division\"]}.ensembl.org/{species}/Location/View?db=core;g={species_info[\"gene_id\"]}\"\n","    return(species_info)\n","\n","with open(\"GeneSymbols_45.txt\", \"r\") as infos:\n","    for line in infos:\n","        gene = line.split(\",\")[0]\n","        current_species = line[:-1].split(\",\")[1]\n","    div = RetrieveDiv(entries)\n","    url = f\"http://{div}.ensembl.org/{entries}/Location/View?db=core;g=YDL168W\"\n","\n","print(url)\n","\n","\n"]},{"cell_type":"code","execution_count":21,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["escherichia_coli_gca_006381815\n","{'gene_symbol': 'HYBE'}\n"]}],"source":["species_info = {}\n","with open(\"GeneSymbols_45.txt\", \"r\") as infos:\n","    for line in infos:\n","        gene = line.split(\",\")[0]\n","        current_species = line[:-1].split(\",\")[1]\n","        species_info[\"gene_symbol\"] = gene\n","\n"]},{"cell_type":"markdown","metadata":{},"source":["## Espèce [deprecated]\n","\n","En fonction de la division\n","Aller chercher : name, display name, accession et division"]},{"cell_type":"code","execution_count":10,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'dict'>\n","dict_keys(['species'])\n","EnsemblBacteria\n"]}],"source":["# !/usr/bin/ python3\n","#-*- coding : utf-8 -*- \n","\n","import requests, sys, json\n"," \n","server = \"https://rest.ensembl.org\"\n","ext = \"/info/species?division=EnsemblBacteria\"\n","divisions = ['EnsemblMetazoa', 'EnsemblFungi', 'EnsemblPlants', 'EnsemblProtists', 'EnsemblVertebrates', 'EnsemblBacteria']\n"," \n","r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n"," \n","if not r.ok:\n","  r.raise_for_status()\n","  sys.exit()\n"," \n","decoded = r.json()\n","print (type(decoded))\n","a= decoded.keys()\n","print(a)\n","print(decoded[\"species\"][1][\"division\"])\n","\n","test_file = open(\"test_species.txt\",\"a\")\n","test_file.write(json.dumps(decoded,indent=2))\n","test_file.close()"]},{"cell_type":"markdown","metadata":{},"source":["## Gène\n","`\"id\": \"ENST` Récupère les ID des transcrits"]},{"cell_type":"code","execution_count":23,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["{'gene_symbol': 'RAD51', 'gene_id': 'ENSG00000051180', 'transcript_id': 'ENST00000527860<br>\\n\\t\\t\\tENST00000423169<br>\\n\\t\\t\\tENST00000557850<br>\\n\\t\\t\\tENST00000382643<br>\\n\\t\\t\\tENST00000267868<br>\\n\\t\\t\\tENST00000645673<br>\\n\\t\\t\\tENST00000525066<br>\\n\\t\\t\\tENST00000526763<br>\\n\\t\\t\\tENST00000532743<br>\\n\\t\\t\\tENST00000531277<br>\\n\\t\\t\\tENST00000533741', 'prot_id': 'ENSP00000432759<br>\\n\\t\\t\\tENSP00000406602<br>\\n\\t\\t\\tENSP00000454176<br>\\n\\t\\t\\tENSP00000372088<br>\\n\\t\\t\\tENSP00000267868<br>\\n\\t\\t\\tENSP00000493712<br>\\n\\t\\t\\tENSP00000431864<br>\\n\\t\\t\\tENSP00000431897<br>\\n\\t\\t\\tENSP00000433924<br>\\n\\t\\t\\tENSP00000436512<br>\\n\\t\\t\\tENSP00000433053'}\n"]}],"source":["# !/usr/bin/ python3\n","#-*- coding : utf-8 -*- \n","import requests, sys, json\n","\n","# Lecture du fichier d'espèce et symboles de gènes\n","with open(\"GeneSymbols_45.txt\", \"r\") as infos:\n","    for line in infos:\n","        symbol = line.split(\",\")[0]\n","        current_species = line[:-1].split(\",\")[1]\n","\n","def InfoGene (species, symbol) :\n","    # Serveur Ensembl pour requête API\n","    server = \"https://rest.ensembl.org\"\n","\n","    # Lien vers l'espèce et symbole de gène courants\n","    ext = f\"/lookup/symbol/{species}/{symbol}?expand=1\" \n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n","    decoded = r.json()\n","\n","    # Retour en cas d'erreur d'info de l'API\n","    if not r.ok:\n","        r.raise_for_status()\n","        sys.exit()\n","\n","    # Construction du dictionnaire contenant les infos fomatées pour le html\n","    species_info = {}\n","    species_info[\"gene_symbol\"] = symbol\n","    species_info[\"gene_id\"] = decoded[\"id\"]\n","\n","    for i in decoded[\"Transcript\"] :\n","        if \"transcript_id\" not in species_info :\n","            species_info[\"transcript_id\"] = i[\"id\"]\n","            species_info[\"prot_id\"] = i[\"Translation\"][\"id\"]\n","        else : \n","            species_info[\"transcript_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"id\"]\n","            species_info[\"prot_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"Translation\"][\"id\"]\n","\n","    return(species_info)\n","\n","print(InfoGene(\"homo_sapiens\",\"RAD51\"))"]},{"cell_type":"markdown","metadata":{},"source":["## Création HTML\n","écrire structure et modifier html"]},{"cell_type":"code","execution_count":22,"metadata":{},"outputs":[],"source":["dic = {'homo_sapiens': [{'symbol': 'RAD51', 'gene_id': 'ENSG00000051180', 'transcript_id': 'ENST00000527860<br>\\n\\t\\t\\tENST00000423169<br>\\n\\t\\t\\tENST00000557850<br>\\n\\t\\t\\tENST00000382643<br>\\n\\t\\t\\tENST00000267868<br>\\n\\t\\t\\tENST00000645673<br>\\n\\t\\t\\tENST00000525066<br>\\n\\t\\t\\tENST00000526763<br>\\n\\t\\t\\tENST00000532743<br>\\n\\t\\t\\tENST00000531277<br>\\n\\t\\t\\tENST00000533741', 'prot_id': 'ENSP00000432759<br>\\n\\t\\t\\tENSP00000406602<br>\\n\\t\\t\\tENSP00000454176<br>\\n\\t\\t\\tENSP00000372088<br>\\n\\t\\t\\tENSP00000267868<br>\\n\\t\\t\\tENSP00000493712<br>\\n\\t\\t\\tENSP00000431864<br>\\n\\t\\t\\tENSP00000431897<br>\\n\\t\\t\\tENSP00000433924<br>\\n\\t\\t\\tENSP00000436512<br>\\n\\t\\t\\tENSP00000433053'}], 'arabidopsis_thaliana': [{'symbol': 'PHYB', 'gene_id': 'AT2G18790', 'transcript_id': 'AT2G18790.2<br>\\n\\t\\t\\tAT2G18790.1', 'prot_id': 'AT2G18790.2<br>\\n\\t\\t\\tAT2G18790.1'}], 'saccharomyces_cerevisiae': [{'symbol': 'SFA1', 'gene_id': 'YDL168W', 'transcript_id': 'YDL168W_mRNA', 'prot_id': 'YDL168W'}], 'Leishmania_major': [{'symbol': 'ACP', 'gene_id': 'LMJF_27_0290', 'transcript_id': 'CBZ12089', 'prot_id': 'CBZ12089'}], 'escherichia_coli_gca_006381815': [{'symbol': 'HYBE', 'gene_id': 'ENSB:fqjn3NbmMv9mSox', 'transcript_id': 'ENSB:fqjn3NbmMv9mSox', 'prot_id': 'ENSB:fqjn3NbmMv9mSox'}]}\n","html = open(\"EMBL.html\",\"w\")\n","\n","## HTML Head\n","head_html = \"\"\"\n","<!DOCTYPE HTML>\n","<html lang=\"fr-FR\">\n","\t<head>\n","\t\t<title> Formulaire javascript 1 </title>\n","\t\t<meta charset=UTF-8/>\n","        <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">\n","\n","        <!-- CSS -->\n","        <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/1.10.20/css/jquery.dataTables.min.css\">\n","        <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/fixedcolumns/3.3.0/css/fixedColumns.dataTables.min.css\">\n","        <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/buttons/1.7.1/css/buttons.dataTables.min.css\">\n","        <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/fixedheader/3.1.9/css/fixedHeader.dataTables.min.css\">\n","    \n","        <!-- JavaScript -->\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://code.jquery.com/jquery-3.7.1.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/1.10.20/js/jquery.dataTables.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/fixedcolumns/3.3.0/js/dataTables.fixedColumns.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/buttons/1.7.1/js/dataTables.buttons.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/buttons/1.7.1/js/buttons.colVis.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/fixedheader/3.1.9/js/dataTables.fixedHeader.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/rowgroup/1.1.2/js/dataTables.rowGroup.min.js\"></script>\n","        \n","        <!-- Script Supplémentaire-->\n","        <script type=\"text/javascript\" class=\"init\">\n","            $(document).ready(function() {\n","                const table = $('#gene').DataTable({\n","                    \"scrollY\": 700,\n","                    \"scrollX\": true,\n","                    lengthMenu: [\n","                        [5, 10, 25, 50, -1],\n","                        [5, 10, 25, 50, 'All']\n","                    ],\n","                    fixedColumns: {\n","                        leftColumns: 2\n","                    },\n","                    fixedHeader: true,\n","                    dom: 'Blfrtip',\n","                    buttons: [\n","                        'colvis'\n","                    ]\n","                });\n","            });\n","            </script>\n","    </head>\n","    <body>\n","        <h1>Resultat du scripting d'aggrégation automatique des annotations</h1>\n","\n","        <table id=\"gene\" class=\"display nowrap cell-border\" style=\"width:100%\">\n","            <thead>\n","                <tr>\n","                    <th><div class=header_1>Species</div></th>\n","                    <th>Gene Symbol</th>\n","                    <th>Gene - EMBL ID</th>\n","                    <th>Protein - EMBL ID</th>\n","                    <th>Transcript - EMBL ID</th>\n","                </tr>\n","            </thead>\n","            <tbody>\n","\"\"\"\n","\n","## HTML Body\n","body_html =\"\"\n","for esp in dic :\n","    champ = dic[esp][0]\n","    body_html += f\"\"\"\n","    <tr>                    \n","        <td><div class=header_1>\n","            {esp}\n","        </div></td>\n","        <td>\n","            {champ[\"symbol\"]}\n","        </td>\n","        <td>\n","            {champ[\"gene_id\"]}\n","        </td>\n","        <td>\n","            {champ[\"prot_id\"]}\n","        </td>\n","        <td>\n","            {champ[\"transcript_id\"]}\n","        </td>\n","    </tr>\"\"\"\n","tail_html = f\"\"\" \n","            </tbody>\n","        </table>\n","    </body>\n","</html>\n","\"\"\"\n","\n","html = open(\"EMBL.html\",\"w\")\n","html.write(head_html[1:]+body_html+tail_html)\n","html.close()\n"]},{"cell_type":"markdown","metadata":{},"source":["## Sauvegarde\n","Y'a des bouts de code utiles  \n","```python  \n","decoded = (json.dumps(decoded,indent=2))  \n","```"]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.13.1"}},"nbformat":4,"nbformat_minor":2}
