{"cells":[{"cell_type":"markdown","metadata":{},"source":["# Essais de requêtes"]},{"cell_type":"markdown","metadata":{},"source":["## Accession\n","Aller chercher un numéro d'accession pour l'espèce qui servira d'entrée pour trouver la BdD ensembl appropriée"]},{"cell_type":"code","execution_count":22,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["{'species': 'homo_sapiens', 'gene_symbol': 'RAD51', 'gene_id': 'ENSG00000051180', 'division': 'www', 'transcript_id': ['ENST00000527860', 'ENST00000423169', 'ENST00000557850', 'ENST00000382643', 'ENST00000267868', 'ENST00000645673', 'ENST00000525066', 'ENST00000526763', 'ENST00000532743', 'ENST00000531277', 'ENST00000533741'], 'prot_id': ['ENSP00000432759', 'ENSP00000406602', 'ENSP00000454176', 'ENSP00000372088', 'ENSP00000267868', 'ENSP00000493712', 'ENSP00000431864', 'ENSP00000431897', 'ENSP00000433924', 'ENSP00000436512', 'ENSP00000433053'], 'gene_browser': 'https://www.ensembl.org/Homo_sapiens/Location/View?db=core;g=ENSG00000051180'}\n","{'species': 'arabidopsis_thaliana', 'gene_symbol': 'PHYB', 'gene_id': 'AT2G18790', 'division': 'plants', 'transcript_id': ['AT2G18790.2', 'AT2G18790.1'], 'prot_id': ['AT2G18790.2', 'AT2G18790.1'], 'gene_browser': 'https://plants.ensembl.org/Arabidopsis_thaliana/Location/View?db=core;g=AT2G18790'}\n","{'species': 'saccharomyces_cerevisiae', 'gene_symbol': 'SFA1', 'gene_id': 'YDL168W', 'division': 'fungi', 'transcript_id': ['YDL168W_mRNA'], 'prot_id': ['YDL168W'], 'gene_browser': 'https://fungi.ensembl.org/Saccharomyces_cerevisiae/Location/View?db=core;g=YDL168W'}\n","{'species': 'Leishmania_major', 'gene_symbol': 'ACP', 'gene_id': 'LMJF_27_0290', 'division': 'protists', 'transcript_id': ['CBZ12089'], 'prot_id': ['CBZ12089'], 'gene_browser': 'https://protists.ensembl.org/Leishmania_major/Location/View?db=core;g=LMJF_27_0290'}\n","{'species': 'escherichia_coli_gca_006381815', 'gene_symbol': 'HYBE', 'gene_id': 'ENSB:fqjn3NbmMv9mSox', 'division': 'bacteria', 'transcript_id': ['ENSB:fqjn3NbmMv9mSox'], 'prot_id': ['ENSB:fqjn3NbmMv9mSox'], 'gene_browser': 'https://bacteria.ensembl.org/Escherichia_coli_gca_006381815/Location/View?db=core;g=ENSB:fqjn3NbmMv9mSox'}\n"]}],"source":["# !/usr/bin/ python3\n","#-*- coding : utf-8 -*- \n","import requests, sys\n","\n","## Fonction pour requête API et sortie en Json\n","def RequestJson (server, ext):\n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n","    \n","    #Retour en cas d'erreur\n","    if not r.ok:\n","        r.raise_for_status()\n","        sys.exit()\n","\n","    return(r.json())\n","\n","## Fonction pour récupérer la division Ensembl en cas d'échec de la méthode principale\n","def RetrieveDivAlt(species) :\n","\n","    # Retrouve les divisions ensembl \n","    server = \"https://rest.ensembl.org\"\n","    ext = \"/info/divisions?\" \n"," \n","    divisions = RequestJson(server,ext)[::-1] # Inverse pour chercher dans les espèce bactériennes d'abord\n","\n","    # Retrouve la division associé à l'espèce\n","    j = 0\n","    div = None\n","    while not div and j<len(divisions) :\n","        \n","        ext = f\"/info/species?division={divisions[j]}\"     \n","        decoded = RequestJson(server,ext)\n","\n","        for i in range(len(decoded[\"species\"])) :\n","            if species in decoded[\"species\"][i][\"name\"]:\n","                div = decoded[\"species\"][i][\"division\"]\n","        j+=1\n","    return(div)\n","\n","## Fonction pour récupérer la division Ensembl \n","def RetrieveDiv (species) :\n","    # Requête des infos concernant l'espèce\n","    server = \"https://rest.ensembl.org\" # Serveur Ensembl pour requête API\n","    ext = f\"/info/assembly/{species}?\" # Requête d'un assemblage de l'espèce\n","    decoded = RequestJson(server, ext) # Résultat requête\n","\n","    accession = decoded['assembly_accession'] # Récupération d'une accession d'un génome    \n","\n","    # Requête des infos concernant l'accession\n","    ext = f\"/info/genomes/assembly/{accession}?\"\n","    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"}) \n","\n","    # Retour en cas d'erreur\n","    if not r.ok:\n","        division = RetrieveDivAlt(species)\n","    else : \n","        division = r.json()[\"division\"] # Résultat API\n","\n","    if division[7:].lower() == \"vertebrates\" :\n","        div = \"www\"\n","    else : \n","        div = division[7:].lower()\n","    \n","    return(div)\n","\n","## Fonction pour récupérer les informations relatives au symbole dans l'espèce spécifiée\n","def InfoGene (species, symbol) :\n","    # Serveur Ensembl pour requête API\n","    server = \"https://rest.ensembl.org\"\n","\n","    # Requête vers l'espèce et symbole de gène courants\n","    ext = f\"/lookup/symbol/{species}/{symbol}?expand=1\"     \n","    decoded = RequestJson(server,ext)\n","    \n","    # Construction du dictionnaire contenant les infos\n","    species_info = {}\n","    species_info[\"species\"] = species\n","    species_info[\"gene_symbol\"] = symbol\n","    species_info[\"gene_id\"] = decoded[\"id\"]\n","    species_info[\"division\"] = RetrieveDiv(species)\n","\n","    for i in decoded[\"Transcript\"] :\n","        if \"transcript_id\" not in species_info :\n","            species_info[\"transcript_id\"] = [i[\"id\"]]\n","            species_info[\"prot_id\"] = [i[\"Translation\"][\"id\"]]\n","        else :\n","            # Formatage pour le HTML \n","            species_info[\"transcript_id\"].append(i[\"id\"]) \n","            species_info[\"prot_id\"].append(i[\"Translation\"][\"id\"])\n","\n","    species_info[\"gene_browser\"] = f\"https://{species_info[\"division\"]}.ensembl.org/{species.capitalize()}/Location/View?db=core;g={species_info[\"gene_id\"]}\"\n","    return(species_info)\n","\n","with open(\"GeneSymbols_45.txt\", \"r\") as infos:\n","    for line in infos:\n","        symbol = line.split(\",\")[0]\n","        current_species = line[:-1].split(\",\")[1]\n","        print(InfoGene(current_species, symbol))\n"]},{"cell_type":"markdown","metadata":{},"source":["## Création HTML\n","écrire structure et modifier html"]},{"cell_type":"code","execution_count":26,"metadata":{},"outputs":[],"source":["dic = {'species': 'homo_sapiens', 'gene_symbol': 'RAD51', 'gene_id': 'ENSG00000051180', 'division': 'www', 'transcript_id': ['ENST00000527860', 'ENST00000423169', 'ENST00000557850', 'ENST00000382643', 'ENST00000267868', 'ENST00000645673', 'ENST00000525066', 'ENST00000526763', 'ENST00000532743', 'ENST00000531277', 'ENST00000533741'], 'prot_id': ['ENSP00000432759', 'ENSP00000406602', 'ENSP00000454176', 'ENSP00000372088', 'ENSP00000267868', 'ENSP00000493712', 'ENSP00000431864', 'ENSP00000431897', 'ENSP00000433924', 'ENSP00000436512', 'ENSP00000433053'], 'gene_browser': 'https://www.ensembl.org/Homo_sapiens/Location/View?db=core;g=ENSG00000051180'}\n","\n","html = open(\"EMBL.html\",\"w\")\n","\n","## HTML Head\n","head_html = \"\"\"\n","<!DOCTYPE HTML>\n","<html lang=\"fr-FR\">\n","\t<head>\n","\t\t<title> Projet API </title>\n","\t\t<meta http-equiv=\"Content-type\" content=\"text/html; charset=utf-8\">\n","        <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">\n","\n","        <!-- CSS -->\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/2.2.1/css/dataTables.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/buttons/3.2.0/css/buttons.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/colreorder/2.0.4/css/colReorder.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/fixedheader/4.0.1/css/fixedHeader.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/fixedcolumns/5.0.4/css/fixedColumns.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/responsive/3.0.3/css/responsive.dataTables.min.css\" rel=\"stylesheet\">\n","        <link type=\"text/css\" href=\"https://cdn.datatables.net/rowreorder/1.5.0/css/rowReorder.dataTables.min.css\" rel=\"stylesheet\">\n","\n","        <!-- JavaScript -->\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://code.jquery.com/jquery-3.7.0.js\"></script> \n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/2.2.1/js/dataTables.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/buttons/3.2.0/js/dataTables.buttons.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/buttons/3.2.0/js/buttons.colVis.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/colreorder/2.0.4/js/dataTables.colReorder.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/fixedcolumns/5.0.4/js/dataTables.fixedColumns.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/fixedheader/4.0.1/js/dataTables.fixedHeader.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/responsive/3.0.3/js/dataTables.responsive.min.js\"></script>\n","        <script type=\"text/javascript\" language=\"javascript\" src=\"https://cdn.datatables.net/rowreorder/1.5.0/js/dataTables.rowReorder.min.js\"></script>\n","\n","        <!-- Script Supplémentaire-->\n","        <script type=\"text/javascript\" class=\"init\">\n","            $(document).ready(function() {\n","                const table = $('#gene').DataTable({\n","                    layout: {\n","                        topStart: {\n","                            buttons: ['colvis']\n","                        }\n","                    },\n","                    fixedHeader: true,\n","                    \"scrollY\": 700,\n","                    \"scrollX\": true,\n","                    lengthMenu: [\n","                        [5, 10, 25, 50, -1],\n","                        [5, 10, 25, 50, 'All']\n","                    ],\n","                    colReorder: {\n","                        columns: ':gt(1)'\n","                    },                                       \n","                });\n","            });\n","            </script>\n","    </head>\n","    <body>\n","        <h1>Resultat du scripting d'aggrégation automatique des annotations</h1>\n","\n","        <table id=\"gene\" class=\"display nowrap cell-border\" style=\"width:100%\">\n","            <thead>\n","                <tr>\n","                    <th><div class=header_1>Species</div></th>\n","                    <th>Gene Symbol</th>\n","                    <th>Gene Browser</th>\n","                    <th>Gene - EMBL ID</th>\n","                    <th>Protein - EMBL ID</th>\n","                    <th>Transcript - EMBL ID</th>\n","                </tr>\n","            </thead>\n","            <tbody>\n","\"\"\"\n","\n","## HTML Body\n","body_html =\"\"\n","# Boucle à travers le fichier d'origine pour créer les lignes du tableau\n","with open(\"GeneSymbols_45.txt\", \"r\") as infos:\n","    for line in infos:\n","        symbol = line.split(\",\")[0]\n","        current_species = line[:-1].split(\",\")[1]\n","        dic= InfoGene(current_species, symbol)        \n","        \n","        link = f\"https://{dic['division']}.ensembl.org/{dic['species'].capitalize()}\" # Raccourci\n","        \n","        # Créer les liens pour les protéines et les transcrits\n","        \n","        transcript = dic[\"transcript_id\"][0]\n","        html_transcript = f\"<a href = {link}/Transcript/Summary?db=core;g={dic[\"species\"]};t={transcript}>{transcript}</a>\"\n","        html_prot = f\"<a href = {link}/Transcript/ProteinSummary?db=core;g={dic[\"species\"]};t={transcript}>{dic[\"prot_id\"][0]}</a>\"\n","\n","        i=1\n","        for i in range(len(dic[\"transcript_id\"][1:])) :\n","            transcript = dic[\"transcript_id\"][i]\n","            html_transcript += f\"<br>\\n\\t\\t\\t<a href = {link}/Transcript/Summary?db=core;g={dic[\"species\"]};t={transcript}>{transcript}</a>\"\n","            html_prot += f\"<br>\\n\\t\\t\\t<a href = {link}/Transcript/ProteinSummary?db=core;g={dic[\"species\"]};t={transcript}>{dic[\"prot_id\"][i]}</a>\"\n","\n","        body_html += f\"\"\"\n","                <tr>                    \n","                    <td><div class=header_1>\n","                        {dic['species']}\n","                    </div></td>\n","                    <td>\n","                        {dic[\"gene_symbol\"]}\n","                    </td>\n","                    <td><a href = {dic[\"gene_browser\"]}>\n","                        View {dic[\"gene_symbol\"]} in gene browser\n","                    </a></td>\n","                    <td><a href = {link}/Gene/Summary?db=core;g={dic['gene_id']}>\n","                        {dic[\"gene_id\"]}\n","                    </a></td>\n","                    <td>\n","                        {html_transcript}\n","                    </td>\n","                    <td>\n","                        {html_prot}\n","                    </td>\n","                </tr>\"\"\"\n","\n","tail_html = f\"\"\" \n","            </tbody>\n","        </table>\n","    </body>\n","</html>\n","\"\"\"\n","\n","html = open(\"EMBL.html\",\"w\")\n","html.write(head_html[1:]+body_html+tail_html)\n","html.close()\n"]},{"cell_type":"markdown","metadata":{},"source":["## Sauvegarde\n","Y'a des bouts de code utiles  \n","```python  \n","decoded = (json.dumps(decoded,indent=2))\n","species_info[\"transcript_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"id\"]\n","species_info[\"prot_id\"] += \"<br>\\n\\t\\t\\t\" + i[\"Translation\"][\"id\"]\n","```"]},{"cell_type":"code","execution_count":15,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["https://www.ensembl.org/Homo_sapiens\n"]}],"source":["dic = {'species': 'homo_sapiens', 'gene_symbol': 'RAD51', 'gene_id': 'ENSG00000051180', 'division': 'www', 'transcript_id': ['ENST00000527860', 'ENST00000423169', 'ENST00000557850', 'ENST00000382643', 'ENST00000267868', 'ENST00000645673', 'ENST00000525066', 'ENST00000526763', 'ENST00000532743', 'ENST00000531277', 'ENST00000533741'], 'prot_id': ['ENSP00000432759', 'ENSP00000406602', 'ENSP00000454176', 'ENSP00000372088', 'ENSP00000267868', 'ENSP00000493712', 'ENSP00000431864', 'ENSP00000431897', 'ENSP00000433924', 'ENSP00000436512', 'ENSP00000433053'], 'gene_browser': 'https://www.ensembl.org/Homo_sapiens/Location/View?db=core;g=ENSG00000051180'}\n","#https://www.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;g=ENSG00000051180;r=15:40694774-40718886;t=ENST00000527860\n","link = f\"https://{dic[\"division\"]}.ensembl.org/{dic[\"species\"].capitalize()}\"\n","print (link)"]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.13.1"}},"nbformat":4,"nbformat_minor":2}
